src/rest/SearchEndpoint.ts

```diff
@@ -396,19 +396,19 @@ export class SearchEndpoint implements ISearchEndpoint {
    * [`QueryBuilder`]{@link QueryBuilder}.
    * @param callOptions An additional set of options to use for this call.
    * @param callParams The options injected by the applied decorators.
-   * @returns {Promise<IPlan>} A Promise of plan results.
+   * @returns {Promise<ExecutionPlan>} A Promise of plan results.
    */
   @path('/plan')
   @method('POST')
   @requestDataType('application/json')
   @responseType('json')
-  public plan(query: IQuery, callOptions?: IEndpointCallOptions, callParams?: IEndpointCallParameters): Promise<IPlan> {
+  public plan(query: IQuery, callOptions?: IEndpointCallOptions, callParams?: IEndpointCallParameters): Promise<ExecutionPlan> {
     const call = this.buildCompleteCall(query, callOptions, callParams);
     this.logger.info('Performing REST query PLAN', query);
 
-    return this.performOneCall<IPlan>(call.params, call.options).then(results => {
-      this.logger.info('REST query successful', results, query);
-      return results;
+    return this.performOneCall<IPlanResponse>(call.params, call.options).then(planResponse => {
```

Comment:
It would be good to convert this line to use `await`. We should also add a `catch` block to handle errors.